### PR TITLE
fix(http-connect): try wrapping fetch so that it's passed and agent t…

### PR DESCRIPTION
**Pull request by MarcMcIntosh**
https://linear.app/prismic/issue/DT-1271/pushing-changes-sometimes-fails-with-error-getaddrinfo-enotfound

Looks like enotfound errors could be coming from node-fetch
https://github.com/node-fetch/node-fetch/issues/449
and
https://github.com/node-fetch/node-fetch/issues/79
seems that lots of requests, looks like people are fixing it by passing a user agent https://github.com/node-fetch/node-fetch/issues/1735

## Checklist before requesting a review
- [ ] I hereby declare my code ready for review.
- [ ] If it is a critical feature, I have added tests.
- [ ] The CI is successful.
- [ ] If there could backward compatibility issues, it has been discussed and planned.

### CODE REVIEW

File: packages/manager/src/lib/fetch.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+const httpAgent = new http.Agent({
+	keepAlive: true,
+});
+const httpsAgent = new https.Agent({
+	keepAlive: true,
+});
+
+const options: RequestInit = {
+	agent: function (_parsedURL) {
```
Comment (bapmrl) [inline packages/manager/src/lib/fetch.ts]: 💡 `_parseUrl` is used so there's no need to append `_`.

**Review by bapmrl — COMMENTED** [review]
⚠️ This fix might work, but it would be reassuring if we could find a way to reproduce the issue 🙂

Also, I think the manager's tests should be using the same `packages/manager/src/lib/fetch.ts` module for consistency.

Note that I found more explanations here: https://github.com/node-fetch/node-fetch/issues/1735.

File: packages/manager/src/lib/fetch.ts (hunk 1 above)
Comment (MarcMcIntosh) [reply packages/manager/src/lib/fetch.ts]: done

**Review by angeloashmore — COMMENTED** [review]
Inline with @bapmrl's comment, it would be good to have a reproduction of the issue or a test that confirms this fix is effective. However, I understand that would be tricky to implement.

The implementation could be simplified, in my opinion, as well as benefit from some context around why the wrapper exists.

I opened a PR with some changes (it was quicker to write the changes than to explain them): https://github.com/prismicio/slice-machine/pull/961

Feel free to use it or not. 🙂

**Pull request merged**